src/modules/new-request-form/fields/LookupField.tsx

```diff
@@ -0,0 +1,245 @@
+import type { IComboboxProps } from "@zendeskgarden/react-dropdowns.next";
+import {
+  Field as GardenField,
+  Label,
+  Hint,
+  Combobox,
+  Option,
+  Message,
+} from "@zendeskgarden/react-dropdowns.next";
+import { useCallback, useEffect, useMemo, useState } from "react";
+import type { Field, FieldOption } from "../data-types";
+import { Span } from "@zendeskgarden/react-typography";
+import SearchIcon from "@zendeskgarden/svg-icons/src/16/search-stroke.svg";
+import debounce from "lodash.debounce";
+import { useTranslation } from "react-i18next";
+import { EmptyValueOption } from "./EmptyValueOption";
+
+function getCustomObjectKey(targetType: string) {
+  return targetType.replace("zen:custom_object:", "");
+}
+
+const EMPTY_OPTION = {
+  value: "",
+  name: "-",
+};
+
+interface LookupFieldProps {
+  field: Field;
+  userId: number;
+  organizationId: string | null;
+  onChange: (value: string) => void;
+}
+
+export function LookupField({
+  field,
+  userId,
+  organizationId,
+  onChange,
+}: LookupFieldProps) {
+  const {
+    id: fieldId,
+    label,
+    error,
+    value,
+    name,
+    required,
+    description,
+    relationship_target_type,
+  } = field;
+  const [options, setOptions] = useState<FieldOption[]>([]);
+  const [selectedOption, setSelectedOption] = useState<FieldOption | null>(
+    null
+  );
+  const [inputValue, setInputValue] = useState<string>(value as string);
+  const [isLoadingOptions, setIsLoadingOptions] = useState<boolean>(false);
+  const { t } = useTranslation();
+
+  const customObjectKey = getCustomObjectKey(
+    relationship_target_type as string
+  );
+
+  const loadingOption = {
+    name: t(
+      "new-request-form.lookup-field.loading-options",
+      "Loading items..."
+    ),
+    id: "loading",
+  };
+
+  const noResultsOption = {
+    name: t(
+      "new-request-form.lookup-field.no-matches.found",
+      "No matches found"
+    ),
+    id: "no-results",
+  };
+
+  const fetchSelectedOption = useCallback(
+    async (selectionValue: string) => {
+      const res = await fetch(
+        `/api/v2/custom_objects/${customObjectKey}/records/${selectionValue}`
+      );
+      const { custom_object_record } = await res.json();
+      const newSelectedOption = {
+        name: custom_object_record.name,
+        value: custom_object_record.id,
+      };
+      setSelectedOption(newSelectedOption);
+      setInputValue(custom_object_record.name);
+      onChange(custom_object_record.id);
```

Comment:
Do we need to call `onChange` here? Since we call this function only from the initial `useEffect` to fetch the initial option, I don't think it is needed.

Reply:
Right, now we don't need this so I'm removing it.